subsys/bluetooth/services/fast_pair/fp_storage/include/fp_storage_pn.h

```diff
@@ -21,14 +21,20 @@ extern "C" {
 #endif
 
 /** Length of buffer used to store Personalized Name. */
+#ifdef CONFIG_BT_FAST_PAIR_STORAGE_EXT_PN_LEN_MAX
```

Comment:
The function `notify_personalized_name` in `fp_gatt_service.c` uses `FP_STORAGE_PN_BUF_LEN` macro. This function compiles regardless of `CONFIG_BT_FAST_PAIR_EXT_PN` being enabled and it will return compile error if it won't find `FP_STORAGE_PN_BUF_LEN` macro.

Reply:
Maybe you know a better solution for this issue?

Reply:
I see. I don't have any idea that would let us get rid of the preprocessor directives. You can consider removing the #ifdef here and moving it in the fp_gatt_service.c:

```
#ifndef FP_STORAGE_PN_BUF_LEN
#define FP_STORAGE_PN_BUF_LEN 0
```

Reply:
It won't work if I understand correctly. We can't remove #ifdef here because we will get error in line:
`#define FP_STORAGE_PN_BUF_LEN (CONFIG_BT_FAST_PAIR_STORAGE_EXT_PN_LEN_MAX + 1)`
when `CONFIG_BT_FAST_PAIR_STORAGE_EXT_PN_LEN_MAX` is undeclared which will happen when `CONFIG_BT_FAST_PAIR_EXT_PN` is not enabled (we include `fp_storage_pn.h` in `fp_gatt_service.c` regardless of Personalized Name extension being enabled or not).

Reply:
We can actually remove #ifdef here and do the mentioned:
`
#ifndef FP_STORAGE_PN_BUF_LEN
#define FP_STORAGE_PN_BUF_LEN 0
`
but it will require to put `#include "fp_storage_pn.h"` under `#if CONFIG_BT_FAST_PAIR_EXT_PN` and as a consequence provide some blank declaration for `fp_storage_pn_get` function that is defined in this header and also used in `fp_gatt_service.c`. I believe our current solution is better than the one I described.

Reply:
Or simply define a helper Kconfig option that is not configurable by the user (e.g. like source inclusion configurations):

```
config BT_FAST_PAIR_STORAGE_PN_LEN_MAX
    default (BT_FAST_PAIR_STORAGE_EXT_PN_LEN_MAX + 1) if BT_FAST_PAIR_STORAGE_EXT_PN
    default 0
```

Reply:
IIRC, we cannot use arithmetic expressions like `(BT_FAST_PAIR_STORAGE_EXT_PN_LEN_MAX + 1)` in Kconfig defaults.

Reply:
The current implementation seems fine to me. The `FP_STORAGE_PN_BUF_LEN` is defined out of Kconfig. If the feature using it is enabled, the buffer size is defined with a Kconfig option (otherwise the buffer is not used so it has size of zero).